### PR TITLE
Sites Management Page: Replace 'Back to My Sites' with 'Back to Sites'

### LINKED ISSUE #67037: Sites Management Page: Replace 'Back to My Sites' with 'Back to Sites'
There are a few instances of 'Back to My Sites' in the codebase:

image

Once #66804 lands, we should change these instances to 'Back to Sites'.

However, it doesn't look like 'Back to Sites' is a translated string yet. Because it impacts the signup flows, we should probably conditionally use 'Back to Sites' when the translation is available, and then remove the condition once all of the translations are complete.

**Done is:**
* All instances of 'Back to My Sites' are changed to 'Back to Sites'

### PR DESCRIPTION
**Pull request by vykes-mac**
#### Proposed Changes

* Replace all instances of `Back to My Sites` with `Back to Sites`. we conditionally replace these since `Back to Sites` translation might not be readily available.

#### Testing Instructions

* Go to `/sites-dashboard` click Add New Sites
* Verify string has` Back to Sites` instead of `Back to My Sites`

##### Before
before

##### After
after

#### Pre-merge Checklist

- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?

Related to #https://github.com/Automattic/wp-calypso/issues/67037

### CODE REVIEW

**Comment by danielbachhuber** [comment]
@vykes-mac Want to add some test instructions? It would be good to run through each of the screens where this text is presented.

**Comment by zaguiini** [comment]
>translation might not be readily available.

Is that a big problem? It will fall back to English, and Calypso is not 100% translated. Chances are that we will forget to remove this conditional and it won't make a lot of sense for the next developer who looks at this file.

File: client/signup/steps/domains/index.jsx
Comment (p-jackson) [inline client/signup/steps/domains/index.jsx]: Maybe it's a bit counter-intuitive, but `hasTranslation` always returns false for `en` users. So when I tested this I still saw the old `'Back to My Sites'` text.

You have to check for English locales as well unfortunately:
```suggestion
		const sitesBackLabelText = englishLocales.includes( this.props.locale )  || i18n.hasTranslation( 'Back to Sites' )
```

Where `englishLocales` is from `@automattic/i18n-utils`

**Review by p-jackson — CHANGES_REQUESTED** [review]
I use this method for changing strings all the time. Yes it is extra work, but I think we owe it to our non-`en` users. As an English-only speaker myself I try to be extra mindful not to assume that everyone will have some passable English skills. After all I have zero competence in any of our other mag16 languages 😅, why would I expect our users to know some English?

That's why we have tools like this at a8c: pxLjZ-60o-p2

>  and Calypso is not 100% translated

I take that as a challenge 😄 

_In saying that ..._

There is another way using the `[String Freeze]` GitHub label p4TIVU-5Jq-p2
Once you're happy that none of the strings in your PR are going to change you apply the `[String freeze]` label which will pre-submit the strings for translation. CI will tell you when translations are complete and you don't merge until they're done.

The label can be a drag when the PR also includes feature changes. Product folks might be very keen to get those changes released and might be happy to compromise on untranslated strings. That's usually when I use `hasTranslations` so that we can ship soon and let translations catch up. But when it's just strings like this, I think `[String freeze]` is better.

---

Request changes because we should either
- Use the `hasTranslations` method, in which case we need to add the English locale check, or
- Simplify the PR, use `[String freeze]`, and hold of merging until CI says translations are complete.

File: client/signup/steps/domains/index.jsx
Comment (vykes-mac) [reply client/signup/steps/domains/index.jsx]: @p-jackson haha I had added the `en` check but forgot to push the changes, that's how I got the text to display correctly on the screenshot.

Comment (vykes-mac) [reply client/signup/steps/domains/index.jsx]: strange all my screenshots and test instructions were not saved 😅

**Comment by vykes-mac** [comment]
@zaguiini How do you feel about creating an Issue in our backlog to clean up the code after translation. We could create the issue after this PR is merge so that we do not forget.

**Review by p-jackson — APPROVED** [review]
I don't think the strings in the `plans` and `site-type` steps are actually testable.
- To see the one in `plans` you'd need to launch a site from the SMP that already had a domain but no plan.
- The `site-type` step isn't used in any of the signup flows.

I think it's still good to change them for completeness. It's possible that down the track we could have a button in the SMP that goes into the site creation flow that's prefilled the domain, which would mean the user goes straight to the plan step. So it's good to be resilient to that sort of thing.

I've added the cleanup task #67183

I've also added some arrows to the images in PR description to make it easier for translators to see.

**Comment by vykes-mac** [comment]
Thank you @p-jackson for creating the cleanup task. I was having issues trying to test the others was gonna modify values while debugging to get it to render so thank you for that feedback.

**Pull request merged**
